### DISCUSSION TITLE
Use Object Storage for raw data

### DISCUSSION THREAD

**Discussion by Wouter0100** [postalserver/postal#1648] [Feature suggestions]
Would it be a suggestion to use S3 for raw data storage? This would greatly reduce the MySQL database size. 

Normally I would say MySQL isn't designed for the storage of larger files/headers, and it would be ideal to write this to a S3 Object Storage storage. This way it would also be possible to store the raw files longer, as now we're mostly constrained by the fact that we don't want to oversize our database servers.

**Comment by willpower232** [comment]
I would say that Postal doesn't want to keep messages or information about them long term, that should be handled by your application which sends messages via Postal.

**Reply by Wouter0100** [reply]
I'd agree, although even the fact that it's stored longer then only in memory or on tmp disk means it's written to MySQL, which I'd personally do not find MySQL the right fit for such kind of file store. Object storage is specifically designed for this.

I've had various problems in the past due to customers saving large blobs in MySQL, where replication would slow down, we hit various other limits when running a larger Galera cluster and had problems with backups due to the large table sizes. Normally I would suggest to keep the table sizes as small as possible, with only the necessary data stored in them.

In this scenario it would also be possible to let the object storage handle the retention, offloading it from Postal.
